sql/core/src/test/scala/org/apache/spark/sql/DatasetAggregatorSuite.scala

```diff
@@ -240,4 +240,15 @@ class DatasetAggregatorSuite extends QueryTest with SharedSQLContext {
     val df2 = Seq(1 -> "a", 2 -> "b", 3 -> "b").toDF("i", "j")
     checkAnswer(df2.agg(RowAgg.toColumn as "b").select("b"), Row(6) :: Nil)
   }
+
+  test("spark-15114 shorter system generated alias names") {
+    val ds = Seq(1, 3, 2, 5).toDS()
+    assert(ds.select(typed.sum((i: Int) => i)).columns.head === "typedsumdouble_c1")
+    val ds2 = ds.select(typed.sum((i: Int) => i), typed.avg((i: Int) => i))
+    assert(ds2.columns.head === "typedsumdouble_c1")
```

Comment:
@cloud-fan Just wanted to show some difference to user between two aggregate expressions like sum(col1), sum(col2) will show up as typedsumdouble_c1 and typedsumdouble_c2. You think its fine to just report without any suffix ? If you think its ok, then may be we can just create resolved Aliases in Column.named as opposed to deferring it to Analyzer ? Please let me know.